source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -530,6 +534,9 @@ Http::FilterDataStatus JsonTranscoderFilter::decodeData(Buffer::Instance& data,
     }
   } else {
     request_in_.move(data);
+    if (checkIfDecoderBufferLimitReached(request_in_.BytesStored())) {
```

Comment:
This is not very accurate.  incoming request bytes are converted on the flight to proto.  It is stored [here](https://github.com/grpc-ecosystem/grpc-httpjson-transcoding/blob/master/src/include/grpc_transcoding/request_message_translator.h#L128) if I read it correctly.

We may have to count its size.

Reply:
Got it. Will spend some time evaluating if there are any other buffers internally.

Reply:
I took a look at the transcoder library. This will only be a problem for the streaming cases. If two streaming messages occur in succession, then the byte limit will be inaccurate and the buffered data will exceed the allowed limit. This should be rare.

For unary request/response, the current implementation works for all cases. The buffer limit is checked before the message reaches any of those internal buffers. So the size unary messages will be counted correctly. 

I would like to move forward with this PR and follow-up later. The diff is getting large. I am not worried about introducing a partial implementation here. There is **never** a case where the transcoder will incorrectly reject a request. It will always count the correct buffer usage or (in the streaming case) undercount buffer usage.